collect_app/src/main/java/org/odk/collect/android/map/BaseLayerSource.java

```diff
@@ -26,16 +30,21 @@
      */
     void showUnavailableMessage(Context context);
 
-    /** Adds any preferences that are specific to this base layer source. */
-    void addPrefs(PreferenceCategory category);
-
     /**
-     * Creates a map fragment configured with a base layer according to the
-     * preference settings.  This method can return null to indicate that
-     * there is no suitable MapFragment implementation available.
+     * Creates an unconfigured map fragment.  This method can return null to
+     * indicate that there is no suitable MapFragment implementation available.
      */
     @Nullable MapFragment createMapFragment(Context context);
 
+    /** Constructs any preferences that are specific to this base layer source. */
+    List<Preference> createPrefs(Context context);
+
+    /** Gets the set of keys for preferences that should be watched for changes. */
+    Collection<String> getPrefKeys();
```

Comment:
This API lets the MapConfigurator handle all the work for listening to preference changes in one place.